src/main/java/Parser.java

```diff
@@ -0,0 +1,69 @@
+import java.util.Scanner;
+
+public class Parser {
+
+    public static Command parse(String inputLine) throws DukeException {
+        Scanner scanner = new Scanner(inputLine);
+        String action = scanner.next();
+        int index;
+        String remainingString;
+        String[] parsedStrings;
+
+        switch (action) {
+            case "list":
+                return new ListCommand();
+            case "bye":
+                return new ExitCommand();
+            case "todo":
+            case "deadline":

```

Comment:
Perhaps you can consider placing 2 comments here to be clear that fallthrough is intentional here?